src/showReferences.ts

```diff
@@ -25,7 +27,7 @@ export class ShowReferencesFeature implements StaticFeature {
 
   constructor(private _client: BaseLanguageClient) {}
 
-  public fillClientCapabilities(capabilities: ClientCapabilities): void {
+  public fillClientCapabilities(capabilities: ClientCapabilities & ExperimentalClientCapabilities): void {
```

Comment:
Curious how add `ExperimentalClientCapabilities` works/helps?

Reply:
`ClientCapabilities` does not define a structure for `experimental` and just defines it as an empty object. 

Setting the client id, will result in an error:
```
capabilities['experimental']['showReferencesCommandId'] = CLIENT_CMD_ID;
    ^^^
    Property 'showReferencesCommandId' does not exist on type '{}'.
```

Combining both interfaces will add some type information for `experimental`. This might not be the most elegant solution, but the only way to extend `ClientCapabilities` for now.